linode/accountavailability/framework_models.go

```diff
@@ -0,0 +1,29 @@
+package accountavailability
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/linode/linodego"
+)
+
+type DataSourceModel struct {
+	Region      types.String `tfsdk:"region"`
+	Unavailable types.Set    `tfsdk:"unavailable"`
+}
+
+func (d *DataSourceModel) parseAvailability(
```

Comment:
If it's not too much trouble, could you also update [the filter data source model](https://github.com/linode/terraform-provider-linode/blob/proj/dc-get-well/linode/accountavailabilities/framework_models.go#L20) to consume this struct and parser?